### PR TITLE
Add Cocoon scheduler docs

### PR DESCRIPTION
**Pull request by CaseyHillers**
An overview of the features supported by the Cocoon scheduler

### CODE REVIEW

File: CI_YAML.md [hunk 1]
```diff
@@ -168,3 +168,14 @@ Cocoon supports tests that are not owned by Flutter infrastructure. By default,
      scheduler: my_external_location
    ```
 8. Send updates to `https://flutter-dashboard.appspot.com/api/update-task-status` - https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/update_task_status.dart
+
+
+## Scheduling Targets
```
Comment (keyonghan) [inline CI_YAML.md:173]: Can we add subsections for pre-submit and post-submit, as they are both scheduled via cocoon now.

Comment (CaseyHillers) [reply CI_YAML.md:173]: Done

File: CI_YAML.md [hunk 2]
```diff
@@ -168,3 +168,27 @@ Cocoon supports tests that are not owned by Flutter infrastructure. By default,
...
+For targets using the Cocoon scheduler, they can run on:
+ * Presubmit (via GitHub checks)
+ * Postsubmit (via [build dashboard](https://flutter-dashboard.appspot.com/#/build))
+
+By default, all targets should use the Cocoon scheduler.
+
+### Presubmit Features
+
+1. GitHub checks enable targets to run immediately, and are available on the pull request page.
+2. Changes to the ci.yaml will be applied during those presubmit runs.
```
Comment (keyonghan) [inline CI_YAML.md:184]: This is blocked on https://github.com/flutter/flutter/issues/108645.
Do you mind provide some context in the above bug re why we switched to use TOT commit? Or it was not intended.

**Review by keyonghan — APPROVED** [review]
LGTM

File: CI_YAML.md (hunk 2 above)
Comment (CaseyHillers) [reply CI_YAML.md:184]: Sorry, I didn't see the comment. I'll fix the bug (I followed up asking for more details as it seems working to me)

**Pull request merged**